packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt

```diff
@@ -212,4 +215,19 @@ class AppContext(
       intent
     )
   }
+
+// region CurrentActivityProvider
+
+  override val currentActivity: AppCompatActivity?
+    get() {
+      val currentActivity = this.activityProvider?.currentActivity ?: return null
+
+      if (currentActivity !is AppCompatActivity) {
+        throw RuntimeException("Current Activity is of incorrect class, expected AppCompatActivity, received ${currentActivity.localClassName}")
+      }
```

Comment:
Maybe `IllegalStateException` is a better fit here? 🤔 
```suggestion
      if (currentActivity !is AppCompatActivity) {
        throw IllegalStateException("Current Activity is of incorrect class, expected AppCompatActivity, received ${currentActivity.localClassName}")
      }
```
or equivalent using Kotlin builtins:
```suggestion
      check(currentActivity is AppCompatActivity) {
        "Current Activity is of incorrect class, expected AppCompatActivity, received ${currentActivity.localClassName}"
      }
```